blockchain/stakeversion_test.go

```
@@ -0,0 +1,805 @@
+		node := newNode(3, height+i, currentNode)
+
+		// Set stake versions.
+		for x := uint16(0); x < params.TicketsPerBlock; x++ {
+			node.voterVersions = append(node.voterVersions, 2)
+		}
+
+		sv, _ := bc.calcStakeVersionByNode(currentNode)
+		node.header.StakeVersion = sv
+
+		currentNode = node
+		bc.bestNode = currentNode
+
+	}
+	height += runCount
+
+	if !bc.isStakeMajorityVersion(0, currentNode) {
+		t.Fatalf("invalid StakeVersion expected 0 -> true")
+	}
+	if !bc.isStakeMajorityVersion(2, currentNode) {
+		t.Fatalf("invalid StakeVersion expected 2 -> true")
+	}
+	if bc.isStakeMajorityVersion(4, currentNode) {
+		t.Fatalf("invalid StakeVersion expected 4 -> false")
+	}
+
+	// Generate 3 intervals with v4 votes and calculate StakeVersion.
+	runCount = interval * 3
+	for i := int64(0); i < runCount; i++ {
+		node := newNode(3, height+i, currentNode)
+
+		// Set stake versions.
+		for x := uint16(0); x < params.TicketsPerBlock; x++ {
+			node.voterVersions = append(node.voterVersions, 4)
+		}
+
+		sv, _ := bc.calcStakeVersionByNode(currentNode)
+		node.header.StakeVersion = sv
+
+		currentNode = node
+		bc.bestNode = currentNode
+	}
+	height += runCount
+
+	if !bc.isStakeMajorityVersion(0, currentNode) {
+		t.Fatalf("invalid StakeVersion expected 0 -> true")
+	}
+	if !bc.isStakeMajorityVersion(2, currentNode) {
+		t.Fatalf("invalid StakeVersion expected 2 -> true")
+	}
+	if !bc.isStakeMajorityVersion(4, currentNode) {
+		t.Fatalf("invalid StakeVersion expected 4 -> true")
+	}
+	if bc.isStakeMajorityVersion(5, currentNode) {
+		t.Fatalf("invalid StakeVersion expected 5 -> false")
+	}
+
+	// Generate 3 intervals with v2 votes and calculate StakeVersion.
+	runCount = interval * 3
+	for i := int64(0); i < runCount; i++ {
+		node := newNode(3, height+i, currentNode)
+
+		// Set stake versions.
+		for x := uint16(0); x < params.TicketsPerBlock; x++ {
+			node.voterVersions = append(node.voterVersions, 2)
+		}
+
+		sv, _ := bc.calcStakeVersionByNode(currentNode)
```

Same here regarding not ignoring error.